src/metabase/driver/generic_sql/util/unprepare.clj

```diff
@@ -14,11 +14,18 @@
   String  (unprepare-arg [this {:keys [quote-escape]}] (str \' (str/replace this "'" (str quote-escape "'")) \')) ; escape single-quotes
   Boolean (unprepare-arg [this _] (if this "TRUE" "FALSE"))
   Number  (unprepare-arg [this _] (str this))
-  Date    (unprepare-arg [this {:keys [iso-8601-fn]}] (first (hsql/format (hsql/call iso-8601-fn (hx/literal (u/date->iso-8601 this)))))))
+  UUID    (unprepare-arg [this _] (str \' this \'))
+  Date    (unprepare-arg [this {:keys [iso-8601-fn]}]
+                            (let [date-exp (u/date->iso-8601 this)]
+                              (first (hsql/format
+                                      (cond
+                                        (fn? iso-8601-fn) (iso-8601-fn date-exp)
+                                        (keyword? iso-8601-fn) (hsql/call iso-8601-fn (hx/literal date-exp))
+                                        :otherwise (hx/literal date-exp)))))))
 
 (defn unprepare
   "Convert a normal SQL `[statement & prepared-statement-args]` vector into a flat, non-prepared statement."
-  ^String [[sql & args] & {:keys [quote-escape iso-8601-fn], :or {quote-escape "\\\\", iso-8601-fn :timestamp}}]
+  ^String [[sql & args] & {:keys [quote-escape iso-8601-fn], :or {quote-escape "\\\\", iso-8601-fn nil}}]
```

Comment:
@jonbender, can you fill me in on why you made the default implementation for this `nil`? It breaks a few databases so just wondering which ones it fixed